### PR TITLE
*: add test cases only support in MySQL 8.0

### PR DESCRIPTION
**Pull request by csuzhangxc**
**NOTE**: I'll update CI pipelines after this PR merged.

### What problem does this PR solve? 

- add MySQL 8.0 into CI, so that we can add some test cases only support in MySQL 8.0

### What is changed and how it works?

- fix some test cases for MySQL 8.0
- add two `ALTER INDEX` test cases 

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: tests/all_mode/run.sh [hunk 1]
```diff
@@ -166,7 +166,7 @@ function run() {
 
     export GO_FAILPOINTS=''
 
-    run_sql_both_source "SET @@GLOBAL.SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'"
+    run_sql_both_source "SET @@GLOBAL.SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION'"
```
Comment (csuzhangxc) [inline tests/all_mode/run.sh:169]: `NO_AUTO_CREATE_USER` removed in MySQL 8.0

**Review by GMHDBJD — COMMENTED** [review]
LGTM

File: tests/shardddl3/conf/single-source-optimistic-2.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+name: test
+task-mode: all
+is-sharding: true
+shard-mode: "optimistic"
+meta-schema: "dm_meta"
+timezone: "Asia/Shanghai"
+ignore-checking-items: ["table_schema"]
```
Comment (lance6716) [inline tests/shardddl3/conf/single-source-optimistic-2.yaml:8]: Didn't know what blocks MySQL 8.0 passing it

File: tests/shardddl3/run.sh [hunk 3]
```diff
@@ -183,6 +183,77 @@ function DM_082() {
...
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(8,'hhh');"
+    run_sql_source2 "insert into ${shardddl1}.${tb3} values(9,'iii');"
+    
+    if [[ "$1" = "pessimistic" ]]; then
+        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "detect inconsistent DDL sequence from source" 1
+    else
+        # schema comparer for optimistic shard DDL can't diff visible/invisible now.
+        # may this needs to be failed?
```
Comment (lance6716) [inline tests/shardddl3/run.sh:237]: seems `show create table` is `  UNIQUE KEY `a` (`a`) /*!80000 INVISIBLE */`, don't know TiDB will use this comment

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
```
[2020-10-22T02:00:28.356Z] [Thu Oct 22 10:00:28 CST 2020] <<<<<< start DM-085 pessimistic >>>>>>

[2020-10-22T02:00:28.356Z] dmctl test cmd: "start-task /home/jenkins/agent/workspace/dm_ghpr_test/go/src/github.com/pingcap/dm/tests/shardddl3/conf/single-source-pessimistic-2.yaml --remove-meta"

[2020-10-22T02:00:31.627Z] dmctl test cmd: "query-status test"

[2020-10-22T02:00:31.627Z] command: query-status test "unit": "Sync" count: 0 != expected: 1, failed the 0-th time, will retry again

[2020-10-22T02:00:33.520Z] ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'index a visible' at line 1

[2020-10-22T02:00:33.521Z] ...
```

**Comment by csuzhangxc** [comment]
> ```
> [2020-10-22T02:00:28.356Z] [Thu Oct 22 10:00:28 CST 2020] <<<<<< start DM-085 pessimistic >>>>>>
> 
> [2020-10-22T02:00:28.356Z] dmctl test cmd: "start-task /home/jenkins/agent/workspace/dm_ghpr_test/go/src/github.com/pingcap/dm/tests/shardddl3/conf/single-source-pessimistic-2.yaml --remove-meta"
> 
> [2020-10-22T02:00:31.627Z] dmctl test cmd: "query-status test"
> 
> [2020-10-22T02:00:31.627Z] command: query-status test "unit": "Sync" count: 0 != expected: 1, failed the 0-th time, will retry again
> 
> [2020-10-22T02:00:33.520Z] ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'index a visible' at line 1
> 
> [2020-10-22T02:00:33.521Z] ...
> ```

I revert CI to MySQL 5.7 for other PRs now. will change to MySQL 8.0 later.

File: tests/shardddl3/conf/single-source-optimistic-2.yaml (hunk 2 above)
Comment (csuzhangxc) [reply tests/shardddl3/conf/single-source-optimistic-2.yaml]: unnecessary, I just copied from other files. removed in https://github.com/pingcap/dm/pull/1171/commits/e93433fb537f9846ee0aee6c0645b5601fdddba6.

File: tests/shardddl3/run.sh (hunk 3 above)
Comment (csuzhangxc) [reply tests/shardddl3/run.sh:237]: in MySQL 8, it also shows as
```
KEY `c2_idx` (`c2`) /*!80000 INVISIBLE */
```

I think this hint will be used only by the optimizer.

**Review by lance6716 — APPROVED** [review]
LGTM

**Pull request merged**